### PR TITLE
test: Add test for -blockversion

### PR DESCRIPTION
**Pull request by maflcko**
`-blockversion` is currently untested, as in: The setting could be made a no-op without any tests failing. Fix that by adding an explicit test for it. Also, related minor cleanups.

### CODE REVIEW

**Comment by naumenkogs** [comment]
Could you point me to where `NORMAL_GBT_REQUEST_PARAMS` is defined?

File: test/functional/mining_basic.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 from test_framework.blocktools import (
     create_coinbase,
+    NORMAL_GBT_REQUEST_PARAMS,
```
Comment (naumenkogs) [inline test/functional/mining_basic.py:16]: Could you point me to where this is defined?

Comment (MarcoFalke) [reply test/functional/mining_basic.py:16]: `git grep NORMAL_GBT_REQUEST_PARAMS` ;)

Comment (naumenkogs) [reply test/functional/mining_basic.py:16]: Oh, I see, I assumed it was introduced in this PR... Resolve pls

File: test/functional/mining_basic.py [hunk 2]
```diff
@@ -55,8 +55,16 @@ def mine_chain(self):
...
         assert_equal(mining_info['currentblocktx'], 0)
         assert_equal(mining_info['currentblockweight'], 4000)
+
+        self.log.info('test blockversion')
+        self.restart_node(0, extra_args=['-mocktime={}'.format(t), '-blockversion=1337'])
+        self.connect_nodes(0, 1)
+        assert_equal(1337, self.nodes[0].getblocktemplate(NORMAL_GBT_REQUEST_PARAMS)['version'])
+        self.restart_node(0, extra_args=['-mocktime={}'.format(t)])
+        self.connect_nodes(0, 1)
+        assert_equal(0x30000000, self.nodes[0].getblocktemplate(NORMAL_GBT_REQUEST_PARAMS)['version'])
```
Comment (guggero) [inline test/functional/mining_basic.py:65]: nit: To me it wasn't obvious at first why the version now is `0x30000000`. Would it make sense to extract this "magic value" into a variable called `versionTaproot` for example?

**Review by guggero — COMMENTED** [review]
ACK fa39c5623545e998ae23d47ef9da63d5ed4dd3cb.

Nice cleanup and great to have the block version covered in a functional test as well!

File: test/functional/mining_basic.py (hunk 2 above)
Comment (MarcoFalke) [reply test/functional/mining_basic.py]: This has nothing to do with taproot. Changed to `VERSIONBITS_TOP_BITS + (1 << VERSIONBITS_DEPLOYMENT_TESTDUMMY_BIT)`

**Review by guggero — COMMENTED** [review]
ACK fa9b48549ca39b862a10bcfd90e3eac2a0e8ad2e.

**Pull request merged**